sdk/storage/AzureStorageBlob/Source/Operations/BlobsOperations.swift

```diff
@@ -0,0 +1,547 @@
+// --------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation. All rights reserved.
+//
+// The MIT License (MIT)
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the ""Software""), to
+// deal in the Software without restriction, including without limitation the
+// rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+// sell copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+//
+// --------------------------------------------------------------------------
+
+import AzureCore
+import Foundation
+
+// swiftlint:disable type_body_length function_body_length
+public final class BlobsOperations {
```

Comment:
Is this all meant to be part of the public API? It seems like it's just used to split out the implementations of client methods for blobs vs containers, and the public API is still meant to be via StorageBlobClient.

Reply:
If it is not public, you cannot do the following:
```swift
client.blobs.list { ... }
```

It will complain that blobs cannot be public `BlobsOperations` is private/internal.

So in that sense, yes, it is intentional.

Reply:
Right, but we still have `client.listBlobs()`, and it seems like we should have only one way to perform a given operation.

Reply:
This would be a decently-sized change from what we have today, and what Android also has - which is, a client with all of the storage-related methods on it. I would prefer to discuss this during an architecture meeting so we can all align on how we want this to work going forward. Regardless, I feel pretty strongly that we should have exactly one way to do any given operation, so whether they all live in the client, or whether they live in sub-clients, we shouldn't have both.